packages/compiler/test/semantic-walker.test.ts

```diff
@@ -29,9 +29,12 @@ describe("compiler: semantic walker", () => {
   function createCollector(customListener?: SemanticNodeListener) {
     const result = {
       models: [] as Model[],
+      exitModels: [] as Model[],
       modelProperties: [] as ModelProperty[],
+      exitModelProperties: [] as ModelProperty[],
       namespaces: [] as Namespace[],
       operations: [] as Operation[],
+      exitOperations: [] as Operation[],
```

Comment:
Is there any reason not to extend the post-order callbacks to every type? Looks like we might still be missing some like `namespaces`, `interfaces`, etc.